### PR TITLE
feature/PUT-016:Added Put operation

### PR DESCRIPTION
**Pull request by Alberto-R25**

### CODE REVIEW

File: src/main/java/com/agilethought/internship/university/controller/UniversityController.java [hunk 1]
```diff
@@ -42,4 +40,15 @@ public UniversityController(UniversityService service){
 		log.info("UniversityController.postCourses - operation response {}", response);
 		return new ResponseEntity<>(response,HttpStatus.CREATED);
 	}
+
+	@PutMapping(value = "/course/{id}")
+	@ResponseStatus(HttpStatus.OK)
+	public ResponseEntity<UpdateCourseResponse> putOperation(
+			@RequestBody UpdateCourseRequest request, @PathVariable String id) {
+		log.info("Calling Put operation request {}", request);
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/controller/UniversityController.java]: It is not a good practice log request information. Please remove this line.

File: src/main/java/com/agilethought/internship/university/controller/UniversityController.java [hunk 2]
```diff
@@ -42,4 +40,15 @@ public UniversityController(UniversityService service){
...
+		UpdateCourseResponse response = service.Updatecourse(request, id);
+		log.info("Calling Put operation response {}", response);
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/controller/UniversityController.java]: Same here, it is not a good practice log response information, add just a singel comment that PUT operation was successful

File: src/main/java/com/agilethought/internship/university/service/UniversityService.java [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-import com.agilethought.internship.university.domain.CourseResponse;
+import com.agilethought.internship.university.domain.*;
 
 import java.util.List;
 
 public interface UniversityService {
 
     CreateCourseResponse createCourse(CreateCourseRequest request);
     List<CourseResponse> getCourses();
+    UpdateCourseResponse Updatecourse(UpdateCourseRequest request, String id);
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/UniversityService.java]: Remember, method names are in camelcase with first letter in lowercase, updateCourse

File: src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java [hunk 4]
```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
         return newList;
     }
 
+    public UpdateCourseResponse Updatecourse(UpdateCourseRequest request, String courseid) {
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java]: Same here about the method name

File: src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java [hunk 5]
```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
...
+
+        UpdateCourseResponse response = new UpdateCourseResponse();
+        Optional<Course> existcourse = repository.findById(courseid);
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java]: Same here, Objects names are in camel case. existCourse

File: src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java [hunk 6]
```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
...
+
+        if (existcourse != null) {
+            Course newcourse = requestToUpdate(request, existcourse.get());
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java]: Same here about object names, newCourse

File: src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java [hunk 7]
```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
...
+            response.setDescription(newcourse.getDescription());
+            response.setImg(newcourse.getImg());
+            response.setStatus(newcourse.getStatus());
+        }
+
+        return response;
+    }
+
+    private Course requestToUpdate(UpdateCourseRequest request, Course savecourse){
+        log.info("Category conversion successful");
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java]: I think this log is not accurate. It says that "conversion was successful" but still lot of code is going to be executed. please move it a the end.

File: src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java [hunk 8]
```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
...
+            response.setTitle(newcourse.getTitle());
+            response.setDescription(newcourse.getDescription());
+            response.setImg(newcourse.getImg());
+            response.setStatus(newcourse.getStatus());
+        }
+
+        return response;
+    }
+
+    private Course requestToUpdate(UpdateCourseRequest request, Course savecourse){
```
Comment (JoseCortes-Agile) [inline src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java]: Remember names are in camel case. saveCourse

**Review by JoseCortes-Agile — CHANGES_REQUESTED** [review]

**Review by JoseCortes-Agile — APPROVED** [review]

**Review by ErnestoZamora — APPROVED** [review]

**Pull request merged**